Sources/NIO/Linux.swift

```diff
@@ -132,5 +137,35 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func countCoreIds(cores: String) throws -> Int {
+        let ids = cores.components(separatedBy: "-")
+        guard
+            let first = ids.first.flatMap(Int.init),
+            let last = ids.last.flatMap(Int.init),
+            last >= first
+        else { throw CpusetError() }
+        return 1 + last - first
+    }
+
+    static func coreCount(cpuset cpusetPath: String) -> Int? {
+        guard
+            let cpuset = try? String(contentsOfFile: cpusetPath, encoding: .utf8).trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: ","),
```

Comment:
I know it's annoying but we'd have to redo the file reading using `NIOFileHandle`'s open/close functions and then probably the `Posix.read` function.

I know this is very annoying and we don't have high-level API to support this, SwiftNIO only offers asynchronous APIs in `NonBlockingFileIO` but those aren't appropriate here :'(